### PR TITLE
feat(messenger): add tor toggle and change daemon option

### PR DESCRIPTION
**Pull request by clegirar**
- [x] Add toggles in devtools that change tor async storage value
- [x] Check tor async storage value and add tor flag (`--tor.mode=required` or `--tor.mode=optional` or `--tor.mode=disabled`) when we run daemon

Signed-off-by: clegirar <clemntgirard@gmail.com>

### CODE REVIEW

File: js/packages/store/context.ts [hunk 1]
```diff
@@ -153,6 +154,10 @@ export type PersistentOptionsDebug = {
 	enable: boolean
 }
 
+export type PersistentOptionsTor = {
+	enable: boolean
```
Comment (Jorropo) [inline js/packages/store/context.ts:158]: Really not the good type.
accepted values are string :
- `disabled`
- `required`
- `optional` (an acceptable default if *unset* isn't suitable)
- *unset* (default)

**Review by Jorropo — CHANGES_REQUESTED** [review]

File: js/packages/components/onboarding/SetupFinished.js [hunk 2]
```diff
@@ -54,6 +54,12 @@ const SetupFinishedBody = () => {
 					enable: false,
 				},
 			})
+			await setPersistentOption({
+				type: PersistentOptionsKeys.Tor,
+				payload: {
+					flag: 'disabled',
```
Comment (Jorropo) [inline js/packages/components/onboarding/SetupFinished.js:60]: Couldn't default to unset ?

File: js/packages/components/settings/DevTools.tsx [hunk 3]
```diff
@@ -344,6 +347,40 @@ const BodyDevTools: React.FC<{}> = () => {
...
+				actionToggle={async () => {
+					await ctx.setPersistentOption({
+						type: PersistentOptionsKeys.Tor,
+						payload: {
+							flag: ctx.persistentOptions?.tor.flag === 'optional' ? 'disabled' : 'optional',
+						},
+					})
+				}}
+				toggled
+			/>
```
Comment (Jorropo) [inline js/packages/components/settings/DevTools.tsx:383]: Couldn't be a dropdown menu instead ?

**Review by Jorropo — CHANGES_REQUESTED** [review]

File: js/packages/components/settings/DevTools.tsx (hunk 3 above)
Comment (clegirar) [reply js/packages/components/settings/DevTools.tsx:383]: We decided that for the beginning to have 2 toggles for the beginning 👍

File: js/packages/components/onboarding/SetupFinished.js (hunk 2 above)
Comment (clegirar) [reply js/packages/components/onboarding/SetupFinished.js:60]: Yep i can 👍

**Review by Jorropo — APPROVED** [review]
SGTM (I can test it to LGTM it if you want)

**Pull request merged**